### PR TITLE
fix: Off by one error in next/previous tab

### LINKED ISSUE #13: nextItemFromIndex suffers off-by-one errors
I found this issue hitting some edge cases in our CI (I assume timing related). On debugging I found an issue where the `nextItemFromIndex` function doesn't return the correct element if it comes to the end of a sequence (off by one error).

https://github.com/Bkucera/cypress-plugin-tab/blob/master/src/index.js#L76-L94

```ts
const seq = [1, 2, 3, 4];

const nextItem = nextItemFromIndex(3, seq, false);
console.log(nextItem); // undefined

const previousItem = nextItemFromIndex(0, seq, true);
console.log(nextItem); // undefined
```

What happens is 3 is _not_ the length of the array, so the  `if (i === seq.length)` is not true. Even if it was, the logic is still wrong, it would have to set `i = -1` instead, because `i` is getting incremented.

The correct implementation should be:
```ts
const nextItemFromIndex = (i, seq, reverse) => {
  let nextIndex;
  if (reverse) {
    nextIndex = i - 1;
    if (nextIndex < 0) {
      nextIndex = seq.length - 1;
    }
  } else {
    nextIndex = i + 1;
    if (nextIndex === seq.length) {
      nextIndex = 0;
    }
  }
  return seq[nextIndex];
}
```

Or a bit more succinct:
```ts
const nextItemFromIndex = (i, seq, reverse) => {
  if (reverse) {
    const nextIndex = i === 0 ? seq.length - 1 : i - 1;
    return seq[nextIndex];
  } else {
    const nextIndex = i === seq.length - 1 ? 0 : i + 1;
    return seq[nextIndex];
  }
}
```

### PR DESCRIPTION
**Pull request by NicholasBoll**
* Fixes #13

This change fixes the edge cases when using shift-tab when on the first focusable element and the edge case when using tab when on the last focusable element. The error was "Cannot call select of undefined"

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -23,4 +23,7 @@
       "autoFix": true
     },
   ],
+  "editor.codeActionsOnSave": {
```
Comment (NicholasBoll) [inline .vscode/settings.json]: The ESLint plugin insisted on this change (added every time I removed it).

File: src/index.js [hunk 2]
```diff
@@ -75,22 +76,14 @@ const performTab = (el, options) => {
...
-      i = seq.length
-    }
-  } else {
-    if (i === seq.length) {
-      i = 0
-    }
+    const nextIndex = i <= 0 ? seq.length - 1 : i - 1
 
+    return seq[nextIndex]
   }
```
Comment (NicholasBoll) [inline src/index.js]: lint rules prevented an `else` here

File: cypress/integration/spec.js [hunk 3]
```diff
@@ -1,4 +1,5 @@
 /// <reference types="cypress"/>
+/* global cy */
```
Comment (NicholasBoll) [inline cypress/integration/spec.js]: `lint-staged` forced me to add this.

File: src/index.js [hunk 4]
```diff
@@ -1,3 +1,4 @@
+/* global Cypress, cy */
```
Comment (NicholasBoll) [inline src/index.js]: `lint-staged` forced me to add this

Comment (Bkucera) [reply src/index.js]: are you sure you are using local install of eslint? the cypress plugin should have these globals set

File: .vscode/settings.json (hunk 1 above)
Comment (Bkucera) [reply .vscode/settings.json]: interesting, I will keep it then

**Comment by Bkucera** [comment]
looks good. I'm going to remove the eslint related changes since I think it's caused by either the cypress plugin not being installed or using global eslint :+1:

**Pull request merged**